shared-internal/src/main/kotlin/io/github/typesafegithub/workflows/shared/internal/GitHubApp.kt

```diff
@@ -0,0 +1,83 @@
+package io.github.typesafegithub.workflows.shared.internal
+
+import com.auth0.jwt.JWT
+import com.auth0.jwt.algorithms.Algorithm
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.plugins.contentnegotiation.ContentNegotiation
+import io.ktor.client.request.header
+import io.ktor.client.request.post
+import io.ktor.serialization.kotlinx.json.json
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.Transient
+import kotlinx.serialization.json.Json
+import java.security.KeyFactory
+import java.security.interfaces.RSAPrivateKey
+import java.security.spec.PKCS8EncodedKeySpec
+import java.time.Instant
+import java.time.ZonedDateTime
+import java.util.Base64
+
+private var cachedAccessToken: Token? = null
+
+@Serializable
+data class Token(
+    val token: String,
+    val expires_at: String,
+    val permissions: Map<String, String>,
+    val repository_selection: String,
+) {
+    @Transient
+    private val expiresAtDate = ZonedDateTime.parse(expires_at)
+
+    fun isExpired() = expiresAtDate.isBefore(ZonedDateTime.now())
+}
+
+private val httpClient =
+    HttpClient {
+        install(ContentNegotiation) {
+            json(
+                Json { ignoreUnknownKeys = false },
+            )
+        }
+    }
+
+suspend fun getInstallationAccessToken(installationId: String = "62885502"): String {
+    if (cachedAccessToken?.isExpired() == false) return cachedAccessToken!!.token
+    cachedAccessToken =
+        httpClient
+            .post("https://api.github.com/app/installations/$installationId/access_tokens") {
+                header("Accept", "application/vnd.github+json")
+                header("Authorization", "Bearer ${generateJWTToken()}")
+                header("X-GitHub-Api-Version", "2022-11-28")
+            }.body()
+    return cachedAccessToken!!.token
+}
+
+private fun generateJWTToken(githubClientId: String = "Iv23liIZ17VJKUpjacBs"): String {
```

Comment:
We can make it nullable:
```suggestion
private fun generateJWTToken(githubClientId: String = "Iv23liIZ17VJKUpjacBs"): String? {
```
and return null if `loadRsaKey()` returns null.